campaign-launcher/interfaceV2/src/components/LaunchCampaign/index.tsx

```diff
@@ -16,16 +17,25 @@ type Props = {
 const LaunchCampaign: FC<Props> = ({ variant, sx }) => {
   const [openCreateCampaignModal, setOpenCreateCampaignModal] = useState(false);
   const [openStakeHmtPromptModal, setOpenStakeHmtPromptModal] = useState(false);
-  const { isConnected } = useAccount();
+
+  const { isConnected, chainId: accountChainId } = useAccount();
+  const { switchChainAsync } = useSwitchChain();
+  const { appChainId } = useNetwork();
   const isXl = useIsXlDesktop();
-  const { stakedAmount } = useStakeContext();
+  const { stakedAmount, isFetchingInfo, isClientInitializing } = useStakeContext();
+  
+  const isDisabled = !isConnected || isFetchingInfo || isClientInitializing;
```

Comment:
Maybe rely just on signer existence? The whole process can be launcher not only if staking is init, but if we have signer and can trigger escrow contract transactions